django/db/backends/base/schema.py

```diff
@@ -1037,11 +1037,28 @@ def _field_indexes_sql(self, model, field):
         return output
 
     def _field_should_be_altered(self, old_field, new_field):
+        old_deconstruct = old_field.deconstruct()
+        new_deconstruct = new_field.deconstruct()
+
         # Don't alter when changing only a field name.
-        return (
-            old_field.column != new_field.column or
-            old_field.deconstruct()[1:] != new_field.deconstruct()[1:]
-        )
+        if old_field.column == new_field.column and old_deconstruct[1:] == new_deconstruct[1:]:
+            return False
+
+        # Don't alter when only db_column changed but column is the same.
+        if old_field.column == new_field.column and old_deconstruct[:3] == new_deconstruct[:3]:
+            old_kwargs = old_deconstruct[3]
+            new_kwargs = new_deconstruct[3]
+            # Attrs only in new_field.
+            changed_kwargs = set(new_kwargs) - set(old_kwargs)
+            # Attrs only in old_field or in both with different values.
+            for old_key, old_value in old_kwargs.items():
+                if old_key not in new_kwargs or old_value != new_kwargs[old_key]:
+                    changed_kwargs.add(old_key)
+            # Only db_column changed and we know field column did not changed.
+            if changed_kwargs == {'db_column'}:
+                return False
+
+        return True
```

Comment:
We've already compare columns in:
```python
old_field.column != new_field.column or
```
so we can simply ignore `db_column` in kwargs, e.g.
```python
def _field_should_be_altered(self, old_field, new_field):
    _, old_path, old_args, old_kwargs = old_field.deconstruct()
    _, new_path, new_args, new_kwargs = new_field.deconstruct()
    # Ignore db_column to not alter when changing only a db_column but it's
    # the same as a field name.
    old_kwargs.pop('db_column', None)
    new_kwargs.pop('db_column', None)
    # Don't alter when changing only a field name.
    return (
        old_field.column != new_field.column or
        (old_path, old_args, old_kwargs) != (new_path, new_args, new_kwargs)
    )
```

Reply:
Great solution! Thanks! I didn't rethink the past implementation. Updated. :+1: